### PR TITLE
Fix default version output type and server type

### PR DESCRIPTION
**Pull request by asaintsever**
# Description

- Fix `--output` description for `glooctl version`: set to `JSON` (while keeping `TABLE` by default for other commands)
- Fix `GlooType` value for server type when deployment is of type `Ingress`

# Context

Users ran into those bugs by using command `glooctl version`

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3381

### CODE REVIEW

**Comment by kdorosh** [comment]
Hey @asaintsever , thanks for the contribution! Can you add a changelog file that will rendered into the release notes? https://github.com/solo-io/go-utils/tree/master/changelogutils for an example see https://github.com/solo-io/gloo/blob/master/changelog/v1.5.0-beta9/stop-copying-k8s-labels.yaml

**Comment by asaintsever** [comment]
Sure I was already working on it :-)

File: changelog/v1.5.0-beta9/fix-outputtype-server-version-type.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+changelog:
+- type: FIX
```
Comment (kdorosh) [inline changelog/v1.5.0-beta9/fix-outputtype-server-version-type.yaml:2]: fix changelog types need the `issueLink` field or else they aren't valid. Is there already an issue open for the bug you are fixing? if not, please open one with the appropriate details. thanks!

**Comment by kdorosh** [comment]
test failure,
```
[It] should override/allow -o flags as expected
  /workspace/gloo/projects/gloo/cli/pkg/cmd/get/output_format_test.go:72

------------------------------
• Failure [0.261 seconds]
Upstream default output should be -o table [It] should override/allow -o flags as expected 
/workspace/gloo/projects/gloo/cli/pkg/cmd/get/output_format_test.go:72

  Expected
      <string>: "[
      {"st..."
  to equal       |
      <string>: "+-----..."

  /workspace/gloo/projects/gloo/cli/pkg/cmd/get/output_format_test.go:75
```

File: projects/gloo/cli/pkg/cmd/version/cmd.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func RootCmd(opts *options.Options, optionsFunc ...cliutils.OptionsFunc) *cobra.
 		Long:    constants.VERSION_COMMAND.Long,
 		PreRunE: func(cmd *cobra.Command, args []string) error {
 			if !cmd.PersistentFlags().Changed(flagutils.OutputFlag) {
-				opts.Top.Output = printers.JSON
+				opts.Top.Output = printers.TABLE
```
Comment (kdorosh) [inline projects/gloo/cli/pkg/cmd/version/cmd.go:40]: we actually depend on this being json in our homebrew formula, see https://github.com/Homebrew/homebrew-core/blob/master/Formula/glooctl.rb#L35

this was intentionally JSON to match `kubectl version`. i think we should leave this as is.

any reason you changed this other than to match our other defaults? might be worth adding a comment to the `printers.JSON` linw here to note why this is the default

Comment (asaintsever) [reply projects/gloo/cli/pkg/cmd/version/cmd.go:40]: Ok but when you type `glooctl version --help` it is stated that default output format is 'table'. So there is some discrepancy here.

Comment (kdorosh) [reply projects/gloo/cli/pkg/cmd/version/cmd.go:40]: Agreed that there is a discrepancy to be addressed; I think we should update the help there to be accurate and leave the command as is. @yuval-k @marcogschmidt thoughts?

Comment (asaintsever) [reply projects/gloo/cli/pkg/cmd/version/cmd.go]: I agree with JSON being the default format for `glooctl version`. I just pushed some changes to fix the usage description: now `--help` shows JSON as default format for `version` command and TABLE for others.

**Pull request merged**